File: haskell/src/Shapes/Client/Shell/Handler.hs
module Shapes.Client.Shell.Handler (handleCircle, handleRectangle, handleSquare) where

import Shapes.Base.Commons
import Shapes.Base.Models
import Shapes.Client.Shell.InputReader (readNumber)
import Text.Printf (printf)

handleCircle :: IO ()
handleCircle = do
    putStrLn "What is the radius of the circle?"
    radius <- readNumber
    let circle = newCircle radius
    printf "Area: %.3f\n" (area circle)
    printf "Perimeter: %.3f\n" (perimeter circle)

handleRectangle :: IO ()
handleRectangle = do
    putStrLn "What is the width of the rectangle?"
    width <- readNumber
    putStrLn "What is the height of the rectangle?"
    height <- readNumber
    let rectangle = newRectangle width height
    printf "Area: %.3f\n" (area rectangle)
    printf "Perimeter: %.3f\n" (perimeter rectangle)

handleSquare :: IO ()
handleSquare = do
    putStrLn "What is the side of the square?"
    side <- readNumber
    let square = newSquare side
    printf "Area: %.3f\n" (area square)
    printf "Perimeter: %.3f\n" (perimeter square)

File: ruby/lib/shapes/client/shell/handler.rb
# frozen_string_literal: true

module Handler
    def self.handle_circle
        puts "What is the radius of the circle?"
        radius = read_number
        circle = Circle.new(radius)
        puts "Area: #{format("%.3f", circle.area)}"
        puts "Circumference: #{format("%.3f", circle.perimeter)}"
    end

    def self.handle_rectangle
        puts "What is the width of the rectangle?"
        width = read_number
        puts "What is the height of the rectangle?"
        length = read_number
        rectangle = Rectangle.new(width, length)
        puts "Area: #{format("%.3f", rectangle.area)}"
        puts "Perimeter: #{format("%.3f", rectangle.perimeter)}"
    end

    def self.handle_square
        puts "What is the side length of the square?"
        side = read_number
        square = Square.new(side)
        puts "Area: #{format("%.3f", square.area)}"
        puts "Perimeter: #{format("%.3f", square.perimeter)}"
    end
end
